Cabal/Distribution/Package.hs

```diff
@@ -175,7 +175,7 @@ instance NFData Module where
 -- @since 2.0
 
 data ComponentId
-    = ComponentId String
+    = ComponentId ShortText
```

Comment:
Well, ComponentId valid characters coincide with what InstalledPackageId used to be: alphanumeric, dash, underscore and period.  isAlphaNum selects Unicode characters, so strictly speaking they are not ASCII. In practice they won't be ASCII if a package name has non-ASCII characters.